src/mbgl/renderer/buckets/symbol_bucket.cpp

```diff
@@ -356,6 +356,27 @@ bool SymbolBucket::hasFormatSectionOverrides() const {
     return *hasFormatSectionOverrides_;
 }
 
+bool SymbolBucket::hasVariableTextAnchors() const {
+    auto hasTextVariableAnchorOffset = [&]() -> bool {
+        auto tvao = layout->get<TextVariableAnchorOffset>();
+        const bool hasAnchorOffset = tvao.match([](const auto&) { return true; });
```

Comment:
Curious, what case this condition covers? When property is not set at all?

Reply:
add isExpression in PossiblyEvaluatedPropertyValue class to check TextVariableAnchorOffset is expression.